logger.go

```diff
@@ -0,0 +1,10 @@
+package scim
+
+import "log/slog"
+
+var log *slog.Logger = slog.Default().WithGroup("scim")
+
+// SetLogger sets the logger for the scim package.
+func SetLogger(l *slog.Logger) {
```

Comment:
This is better, as it allows passing a custom `*slog.Logger` instance. But what if the users of this package are using [zerolog](https://github.com/rs/zerolog) or [zap](https://github.com/uber-go/zap)?